opts/envdir.go

```diff
@@ -0,0 +1,43 @@
+package opts
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
+
+/*
+Read in a directory with files acting as environment variables.
+*/
+func ParseEnvDir(directory string) ([]string, error) {
+	dir, err := ioutil.ReadDir(directory)
+	if err != nil {
+		return []string{}, err
+	}
+
+	lines := []string{}
+	for _, info := range dir {
+		// ignore directories
+		if !info.IsDir() {
+			variable := info.Name()
+			if strings.ContainsAny(variable, whiteSpaces) {
```

Comment:
So I mostly agree, but I feel like doing that is out of scope of this function as well as out of scope for this PR.

If I were to implement more strict validation, I'd do it later after the collection of all the values that way even CLI args passed in through `-e FOO=bar` style were all validated the same way.

The role of `ParseEnvDir` should do it's one thing, and that's crawl a directory and convert into `k=v` pairs, which get passed along later down the pipeline.